src/layers/common/dropout.rs

```diff
@@ -0,0 +1,126 @@
+//! Applies a Dropout layer to the input data `x`
+//!
+//! The variables are:
+//!
+//! - `y`: output value
+//! - `x`: input value
+//! - `p`: dropout probability
+
+use super::FilterLayer;
+use capnp_util::*;
+use co::{IBackend, SharedTensor};
+use conn;
+use layer::*;
+use juice_capnp::dropout_config as capnp_config;
+use std::rc::Rc;
+use util::ArcLock;
+
+#[derive(Debug, Clone)]
+/// [Dropout](./index.html) Layer
+pub struct Dropout<T, B: conn::Dropout<T>> {
+    probability: f32,
```

Comment:
why is `probability` duplicated here and in `DropoutConfig`?

Reply:
doesn't make sense, I am going to look into it, pretty sure this one is superflous